c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -583,6 +697,15 @@ class DeviceCachingAllocator {
       process_events();
     }
 
+    // try to allocate according to plan
+    if (memory_planner.active) {
+      Block* plan_block = memory_planner.malloc(size, device, stream);
+      // nullptr indicates event is not allocated according to plan
+      if (plan_block) {
+        return plan_block;
+      }
+    }
```

Comment:
This is OK for now to unblock you, but it might be more reasonable to keep existing allocator intact and make the new allocator the entry point within the tracking context. And the new allocator can call into the old one if it fails o allocate blocks.